tests/tests_test.go

```diff
@@ -0,0 +1,278 @@
+				}
+				for _, p := range podList.Items {
+					if p.Status.Phase == "Running" {
+						podCmd := pods.PodCommand{
+							Pod:     &p,
+							RestCfg: sourceCfg,
+							Args:    []string{"sh", "-c", "dd if=/dev/zero of=/data/test/file_1 bs=60M count=1"},
+						}
+						err = podCmd.Run()
+						if err != nil {
+							log.Println(err, "Failed running ls command inside destination Pod",
+								"pod", path.Join(p.Namespace, p.Name),
+								"command", "dd if=/dev/zero of=/data/test/file_1 bs=60M count=1;")
+						}
+					}
+					return string(p.Status.Phase)
+				}
+				return ""
+			}, time.Minute*5, time.Second).Should(Equal("Running"))
+		})
+
+		AfterEach(func() {
+			for _, ns := range namespaces {
+				err = sourceClient.CoreV1().Namespaces().Delete(ctx, ns, metav1.DeleteOptions{})
+				if err != nil {
+					log.Println(err)
+				}
+				err = hostClient.Delete(context.TODO(), &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns}})
+				if err != nil {
+					log.Println(err)
+				}
+			}
+		})
+
+		It("Should create a new migration", func() {
+			By("Creating a new migmigration")
+
+			migrationName := "foo"
+			migration := NewMigMigration(migrationName, migPlanName, false, false)
+			Expect(hostClient.Create(ctx, migration)).Should(Succeed())
+
+			fooMigration := &migapi.MigMigration{}
+
+			Eventually(func() string {
+				dvms := &migapi.DirectVolumeMigrationList{}
+				err := hostClient.List(ctx, dvms, client.InNamespace(MigrationNamespace))
+				if err != nil {
+					log.Println(err)
+				}
+				for _, dvm := range dvms.Items {
+					if dvm.GenerateName == migrationName+"-" {
+						if dvm.Status.Itinerary == "VolumeMigration" {
+							dvmps := &migapi.DirectVolumeMigrationProgressList{}
+							err = hostClient.List(ctx, dvmps, client.MatchingLabels{"directvolumemigration": string(dvm.UID)})
+							if err != nil {
+								log.Println(err)
+							}
+							for _, dvmp := range dvmps.Items {
+								if dvmp.Status.TotalProgressPercentage == "100%" {
+									return dvm.Status.Phase
+								}
+							}
+						} else {
+							log.Println("Direct volume migration Failed")
+						}
+					}
+				}
+				return ""
+			}, time.Minute*5, time.Second).Should(Equal("Completed"))
+			Eventually(func() string {
+				hostClient.Get(ctx, client.ObjectKey{Name: migrationName, Namespace: MigrationNamespace}, fooMigration)
+				if fooMigration.Status.Itinerary == "Final" {
+					return fooMigration.Status.Phase
+				}
+				return ""
+			}, time.Minute*5, time.Second).Should(Equal("Completed"))
+
+			// TODO check the destination pvc for data
+			Eventually(func() bool {
+				podList := &v1.PodList{}
+				hostClient.List(ctx, podList, &client.ListOptions{
+					LabelSelector: labels.SelectorFromSet(map[string]string{
+						"app": "longpvc-test",
+					},
+					),
+					Namespace: "ocp-41583-longpvcname",
+				})
+				destinationFile := []string{}
+				for _, p := range podList.Items {
+					podCmd := pods.PodCommand{
```

Comment:
question: do we need to run this command on a pod list? Should we be running on this on the first pod returned from the list? Same for source below

Reply:
@alaypatel07 There is always gonna be one pod in the list since we are fetching it with labels. So what difference would it make? If any of the pods (source or destination) is not present then the check will fail. I don't know what the standard practice is though, just curious. We can break the loop after the first iteration if needed.

Reply:
so think about this, Kube is eventually consistent, i.e. there can be situations where there are two replicas temporarily. If that happens the best-case scenario would be, your code will overwrite the result of second with first. The worst-case scenario would be, if one of the two pods are in a terminating state (because kube terminates and is trying to move to a consistent state, this will error out.)

Reply:
Cool! Thanks for clearing out the doubt I had!